### PR TITLE
[styles] add cairn

### LINKED ISSUE #6979: Display man_made=cairn
this is a pretty important thing for hiking navigation, and should be reasonably prominently rendered in the outdoors style. 

it might even be good to treat named ones the same as place=locality in search.

icon:

### PR DESCRIPTION
**Pull request by RedAuburn**
fixes #6979

icon:

image1

### CODE REVIEW

**Comment by RedAuburn** [comment]
ah, forgot search aliases

**Comment by RedAuburn** [comment]
at the moment, we can't distinguish cairn sizes, however i've found [cairn=small](https://taginfo.openstreetmap.org/keys/cairn#values), so we could add smaller rendering for that & then point people to it if they think cairns are rendering too large

**Comment by pastk** [comment]
> at the moment, we can't distinguish cairn sizes, however i've found [cairn=small](https://taginfo.openstreetmap.org/keys/cairn#values), so we could add smaller rendering for that & then point people to it if they think cairns are rendering too large

This tag is almost not used, so we can't rely on it.

I've checked and looks like micromapping of waymark cairns is not uncommon, e.g. see around 54.6478124,-3.1476357.

So I suggest to add them at z19- in the main style and z18- in the outdoors for now...

**Comment by RedAuburn** [comment]
now rendering z19 in clear, z18 in outdoor

the icon is the same colour as cave, guidepost etc. i don't think it should be light, they are important landmarks for hiking

**Comment by pastk** [comment]
sorry by "light" I meant "visually light" :) like our previous cave icon was very "heavy/massive" looking and now its much "lighter"

File: data/styles/outdoors/include/Icons.mapcss [hunk 1]
```diff
@@ -256,6 +256,8 @@ node|z16-[tourism=viewpoint],
 
 
 /* 5.5 Others */
+node|z18-[man_made=cairn],
+{text: name;text-color: @poi_label;text-offset: 1;font-size: 10;}
```
Comment (pastk) [inline data/styles/outdoors/include/Icons.mapcss:260]: I think just put both icon and caption declaration in one line, atm you duplicate some params like `font-size`.
Also in outdoors we use slightly bigger font sizes, for this zoom it should be `12`.

**Comment by RedAuburn** [comment]
> sorry by "light" I meant "visually light" :) like our previous cave icon was very "heavy/massive" looking and now its much "lighter"

ah ok 👍️ I think it's alright because it only displays at quite high zooms, and tends to be one of the only features around (up hills, in fields etc). If we render it at a lower zoom level, we can make a lighter version for it.

File: data/styles/outdoors/include/Icons.mapcss (hunk 1 above)
Comment (RedAuburn) [reply data/styles/outdoors/include/Icons.mapcss:260]: fixed

**Comment by pastk** [comment]
Looks like you didn't pull in the latest commits?

**Review by pastk — APPROVED** [review]
Thanks!

File: data/categories.txt [hunk 2]
```diff
@@ -15872,3 +15872,16 @@ uk:Колекціонер
 vi:Nhà sưu tập
 zh-Hans:收藏家
 zh-Hant:集電極
+
+man_made-cairn
+en:4Cairn
```
Comment (biodranik) [inline data/categories.txt:15877]: Should it really be searchable? Who will guess that Cairn is supported by OM, and tries to type its name properly in the search?

**Review by biodranik — COMMENTED** [review]
If cairns are mapped, then the trail is also highly likely already mapped, right?

File: data/categories.txt (hunk 2 above)
Comment (RedAuburn) [reply data/categories.txt:15877]: assuming cairns with names will still be searchable without a category type name, i'll remove these strings 👍️

**Comment by RedAuburn** [comment]
> If cairns are mapped, then the trail is also highly likely already mapped, right?

yes, but cairns (at least in the uk) are sometimes named and treated the same as a locality, eg. "dead wife's grave" in my screenshot above)

**Comment by RedAuburn** [comment]
removed categories, ptal @biodranik :)

File: data/categories.txt (hunk 2 above)
Comment (pastk) [reply data/categories.txt]: I'd keep them searchable. Cairns are quite often erected as major "milestones" e.g. on a mountain pass and e.g. in ex-USSR mountaineering groups traditionally leave there written notes to future adventurers. And they don't necessary have a name.

Comment (biodranik) [reply data/categories.txt]: @vng if objects are not added to categories.txt, are they searched by their name anyway?

**Comment by RedAuburn** [comment]
> pastk: I'd keep them searchable. Cairns are quite often erected as major "milestones" e.g. on a mountain pass and e.g. in ex-USSR mountaineering groups traditionally leave there written notes to future adventurers. And they don't necessary have a name.

due to this, i think it makes sense to keep them searchable.

it's also possible that hikers may want to visit all the cairns in the area, so can plan a route to visit them by showing them in the search results

**Review by pastk — APPROVED** [review]

**Comment by pastk** [comment]
Tests fail
```
indexer/categories_holder.cpp:231
CHECK(langCode != kUnsupportedLocaleCode) Invalid language code: af at line: 15878
```
Somehow we don't have `af` in categories, but we do have it in types_strings.txt and strings.txt :/

**Comment by RedAuburn** [comment]
ah oops, fixed 👍️

**Comment by pastk** [comment]
> 
> Somehow we don't have `af` in categories, but we do have it in types_strings.txt and strings.txt :/

I've mentioned it in https://github.com/organicmaps/organicmaps/pull/7012#issuecomment-1872773302

File: data/categories.txt [hunk 3]
```diff
@@ -15872,3 +15872,16 @@ uk:Колекціонер
...
+de:Steinmännchen
+et:Karjäär
+ja:ケルン
+nl:Steengroeve
+pl:Kopiec
+pt:Moledro ou mariola
+pt-BR:Moledro ou mariola
+ru:Тур
```
Comment (vng) [inline data/categories.txt:15885]: - Do we need _carn_ in categories?
- "Тур" for ru and uk seems wrong. @pastk @biodranik Do you know this word's meaning other than _bull_?

Comment (pastk) [reply data/categories.txt:15885]: Please check discussion in https://github.com/organicmaps/organicmaps/pull/6981#discussion_r1437273347

"Тур" in russian is exactly "cairn" i.e. a pyramid of rocks. Its used primarily by hikers / mountaineers / alpinists.

**Review by vng — APPROVED** [review]

**Comment by RedAuburn** [comment]
rebased @vng :)

**Pull request merged**